### PR TITLE
[IMP] <base_report_auto_create_qweb> added button to fix #255

### LINKED ISSUE #255: [8.0] base_report_auto_create_qweb - it does not create qweb views
Tested on runbot.

When creating a new report it seems that it does not create the views. So you can not configure a new one.

### PR DESCRIPTION
**Pull request by oihane**

### CODE REVIEW

File: base_report_auto_create_qweb/models/report_xml.py [hunk 1]
```diff
@@ -9,6 +9,12 @@
 class IrActionsReport(models.Model):
     _inherit = 'ir.actions.report.xml'
 
+    def _format_template_name(self, text):
```
Comment (pedrobaeza) [inline base_report_auto_create_qweb/models/report_xml.py]: You can use slugify library (already a requirement) for doing this: https://github.com/odoo/odoo/blob/8.0/addons/website/models/website.py#L87

**Comment by pedrobaeza** [comment]
The code that is triggered on create/write is still there. If we put the button, the idea is to always make it manually, doesn't it?

**Comment by oihane** [comment]
Copy works fine as it is, if you remove that code it will not make a copy of the related views.

**Comment by pedrobaeza** [comment]
:+1:

**Comment by oihane** [comment]
I found another bug :worried:
if no suffix is provided when copying it fails, I'm currently fixing it

**Comment by oihane** [comment]
@tafaRU @eLBati can you both check this again... I think it works everything fine now...

**Comment by agaldona** [comment]
Functionally tested :+1:

File: base_report_auto_create_qweb/models/report_xml.py [hunk 2]
```diff
@@ -9,6 +9,12 @@
...
+        from unidecode import unidecode
```
Comment (eLBati) [inline base_report_auto_create_qweb/models/report_xml.py]: Please read https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#external-dependencies

File: base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            'model': 'res.partner',
+            'report_type': 'qweb-html',
+            'report_name': 'test1.report_test',
+        })
+        report_html.button_create_qweb()
+        view_num = self.view_model.search_count(
+            [('name', 'ilike', report_html.report_name.split('.')[1]),
+             ('type', '=', 'qweb')])
+        self.assertEqual(view_num, 1, 'Only one view must be created.')
+        self.assertNotEqual(view_num, 0, 'There are not related views')
```
Comment (eLBati) [inline base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py:28]: As `view_num` has already been checked, this line is unnecessary

File: base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            'model': 'res.partner',
+            'report_type': 'qweb-pdf',
+            'report_name': 'test2.report_test',
+        })
+        report_pdf.button_create_qweb()
+        view_num = self.view_model.search_count(
+            [('name', 'ilike', report_pdf.report_name.split('.')[1]),
+             ('type', '=', 'qweb')])
+        self.assertEqual(view_num, 1, 'One view must be created.')
+        self.assertNotEqual(view_num, 0, 'There are not related views.')
```
Comment (eLBati) [inline base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py:42]: Same as above

File: base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py (hunk 3 above)
Comment (oihane) [inline base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py:28]: And what about changing the order? :wink:

Comment (eLBati) [inline base_report_auto_create_qweb/tests/test_base_report_auto_create_qweb.py:28]: If you want a specific error message when `view_num` == 0, fine

**Comment by eLBati** [comment]
Thanks @oihane,

here are my tests.
If I create a new report setting
template name = invoice3
and click 'Create QWeb view'
I get

```
File "/srv/openerp/instances/openerp-oca-runbot/parts/odoo-extra/runbot/static/build/3115828-260-6d8345/openerp/addons/base_report_auto_create_qweb/models/report_xml.py", line 113, in button_create_qweb
    report_name = self.report_name.split('.')[1]
IndexError: list index out of range
```

I suggest to raise a UserError

Then, If I set
name = invoice3
model = account.invoice
report type = PDF
template name = account.invoice3
and click 'Create QWeb view',

When I open an invoice and click on
print -> invoice3
I get

```
QWebTemplateNotFound: Template u'account.invoice3' not found
```

**Comment by oihane** [comment]
@eLBati fixed all reported issues, ready for another check :wink:

**Comment by anajuaristi** [comment]
Functionally :+1: RTM

**Pull request merged**